src/libs/OptionsListUtils.js

```diff
@@ -365,6 +367,36 @@ function getAllReportErrors(report, reportActions) {
     return allReportErrors;
 }
 
+/**
+ * Get the last message text from the report directly or from other sources for special cases.
+ * @param {Object} report
+ * @returns {String}
+ */
+function getLastMessageTextForReport(report) {
+    let lastMessageTextFromReport = '';
+    if (ReportUtils.isReportMessageAttachment({text: report.lastMessageText, html: report.lastMessageHtml})) {
+        lastMessageTextFromReport = `[${Localize.translateLocal('common.attachment')}]`;
+    } else {
+        lastMessageTextFromReport = report ? report.lastMessageText || '' : '';
+
+        // Yeah this is a bit ugly. If the latest report action that is not a whisper has been moderated as pending remove, then set the last message text to the text of the latest visible action that is not a whisper.
+        const lastNonWhisper =
+            _.find(allSortedReportActions[report.reportID], (action) => {
+                const isWhisper = (action.whisperedTo || []).length > 0;
```

Comment:
Can we just create an isWhisperReportAction in ReportActionUtils and then use it here and below to dry this up a bit?

Reply:
Sure. We can use it elsewhere too.